GitCommands/RevisionReader.cs

```diff
@@ -395,23 +398,21 @@ long ParseUnixDateTime(ref ReadOnlySpan<byte> array)
             var committer = reader.ReadLine();
             var committerEmail = reader.ReadLine();
 
-            (string? subject, string? body, bool hasMultiLineMessage) = reader.ReadToEndOfBody(hasPathFilter);
+            bool skipBody = sixMonths > authorUnixTime;
+            (string? subject, string? body, bool hasMultiLineMessage) = reader.ReadToEndOfBody(hasPathFilter, skipBody);
+
+            // We keep a full multiline message body within the last six months.
+            // Note also that if body and subject are identical (single line), the subject never need to be stored
+            skipBody = skipBody || !hasMultiLineMessage;
 
-            if (author is null || authorEmail is null || committer is null || committerEmail is null || subject is null || body is null)
+            if (author is null || authorEmail is null || committer is null || committerEmail is null || subject is null || (!skipBody && body is null))
```

Comment:
❓
```suggestion
            if (author is null || authorEmail is null || committer is null || committerEmail is null || subject is null || (skipBody != (body is null)))
```

Reply:
This is on the limit for pattern also for me, but changing as I assume I will prefer this in some time...